api/proto/attest/attest.proto

```diff
@@ -22,38 +23,135 @@ message ZAttestResponse {
   ZAttestRespType respType = 1;    //type of the response
   ZAttestNonceResp nonce = 2;      //nonce from Controller
   ZAttestQuoteResp quoteResp = 3;  //attest quote response from Controller
+  AttestInfoResp  info_resp = 4;   //attest info response from Controller
 }
 
 enum ZAttestReqType {
   ATTEST_REQ_NONE = 0;
   ATTEST_REQ_CERT = 1;   //EVE X.509 certificates
   ATTEST_REQ_NONCE = 2;  //nonce request to Controller
   ATTEST_REQ_QUOTE = 3;  //quote msg
+  Z_ATTEST_REQ_TYPE_INFO = 4; //info msg
 }
 
 enum ZAttestRespType {
   ATTEST_RESP_NONE = 0;
   ATTEST_RESP_CERT = 1;       //response to cert msg
   ATTEST_RESP_NONCE = 2;      //response to quote request
   ATTEST_RESP_QUOTE_RESP = 3; //response to quote msg
+  Z_ATTEST_RESP_TYPE_INFO = 4; //response to quote msg
 }
 
 message ZAttestNonceResp {
   bytes nonce = 1;      //nonce to use in quote generation
 }
 
+//Hash algo used in digests, and PCR Extend operations
+enum TpmHashAlgo {
+  TPM_HASH_ALGO_INVALID = 0;
+  TPM_HASH_ALGO_SHA1 = 1;
+  TPM_HASH_ALGO_SHA256 = 2;
+  TPM_HASH_ALGO_SHA512 = 3;
+}
+
+//A hash algo, and corresponding digest derived using that algo
+message TpmEventDigest {
+   TpmHashAlgo hash_algo = 1;
+   bytes digest = 2;
+}
+
+//Taken from
+//5.2 Crypto Agile Log Entry Format, from https://trustedcomputinggroup.org/wp-content/uploads/EFI-Protocol-Specification-rev13-160330final.pdf
+message TpmEventLogEntry {
+   uint32 index = 1;             // Sequential index of event in the log
+   uint32 pcr_index = 2;         // PCR index to which this event was measured
+   uint32 event_type = 3;        // The type of this event
+   TpmEventDigest digest = 4;    // The digest corresponding to this event
+   bytes  event_data_binary = 5; // The raw event data bytes
+   string event_data_string = 6; // Textual representation of the event data
+}
+
+enum AttestGPSInput {
+   ATTEST_GPS_INPUT_INVALID = 0;
+   ATTEST_GPS_INPUT_PRESENT = 1; //GPS Co-ordinates are available
+   ATTEST_GPS_INPUT_ABSENT  = 2; //GPS Co-ordineates are not available/reliable
+}
+
+message AttestGPSCoordinates {
+   AttestGPSInput gps_input = 1; // Are the co-ordinates populated ?
+   double latitude = 2;          // Latitude of the device geo-location
+   double longitude = 3;         // Longitude of the device geo-location
+}
+
+enum AttestVersionType {
+   ATTEST_VERSION_TYPE_INVALID = 0;
+   ATTEST_VERSION_TYPE_EVE = 1;      // BaseOS version
+   ATTEST_VERSION_TYPE_FIRMWARE = 2; // Firmware version (e.g. UEFI)
```

Comment:
We can get the BIOS version today. Isn't that a different string than the UEFI version? If so add a value for BIOS version.

Reply:
@eriknordmark it is the same.